### ISSUE TITLE
[bug] conans.tools.Version() lose the last digit of patch version if there is a tweak number

### ISSUE THREAD

**Issue by SpaceIm** [conan-io/conan#10000]
### Environment Details (include every applicable attribute)
  * Operating System+version: Macos Big Sur
  * Compiler+version: Doesn't matter
  * Conan version: 1.42.1
  * Python version: 3.9.7

### Steps to reproduce (Include if Applicable)

```python
from conans import tools

print(tools.Version("1.2.189.2").patch)
print(tools.Version("1.2.18.2").patch)
print(tools.Version("1.2.1.2").patch)
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

output:
18 (instead of 189)
1 (instead of 18)
1

In the meantime, `print(tools.Version("1.2.189").patch)` properly displays 189.

**Comment by memsharded** [comment]
This seems a very unfortunate bug on the node-semver python library implementation. Might be fixed in newer versions, but we were stuck there because apparently the library was changing behavior (it can do it, because it is 0.X, not stable yet, so really not to blame them). I don't think this can be fixed in 1.X

We have started to introduce a new ``Version`` class in 2.0 to replace the ``node-semver`` dependency, it will probably be easier to maintain and more flexible for our users.

**Comment by memsharded** [comment]
New ``Version`` class fixing this merged to ``develop2``, will be in 2.0-alpha2, this can be closed.
